### PR TITLE
Remove needs db and add patch

### LINKED ISSUE #3436: Skip `needs_db` and always map ES hits to DB results
## Problem

Currently the serializers and request params can opt-in and out of mapping ES hits to DB results using the `needs_db` param. This was added when the DB mapping was introduced as a way to test the functionality's viability and peformance impact (although it was not used as such).

## Description

Now that our indices are leaner and contain no info beyond what's indexed and searchable the flag is almost always set to `True` and hence does not need to exist at all. Remove all occurrences of `needs_db` and replace it with a code branch that is always active.

This will simplify a lot of logic and clarify that there is only one path taken by the API from ES hits to JSON response.

### PR DESCRIPTION
**Pull request by ngken0995**
## Fixes

Fixes #3436 by @dhruvkb 

## Description

Remove `needs_db` because ES hit will alway map to DB result. Creating expected data by storing them in pickle file and retrieving those data in testing with open file and use patch to mock `get_db_result` return value.

## Testing Instructions

`just api/test`

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: api/api/views/media_views.py [hunk 1]
```diff
@@ -1,6 +1,5 @@
 import logging
 from typing import Union
-
```
Comment (dhruvkb) [inline api/api/views/media_views.py:3]: That's weird! This separator between imports was justified!

**Review by dhruvkb — CHANGES_REQUESTED** [review]
In `test_auth.py` we're not testing for the contents of the results so for those tests, mocking `get_db_results` to return the input argument `results` as-is would've worked.

File: api/api/views/media_views.py (hunk 1 above)
Comment (ngken0995) [reply api/api/views/media_views.py:3]: How can a mock retrieve the input and return input? is in this documentation related to unittest? https://docs.python.org/3/library/unittest.mock.html

Comment (dhruvkb) [reply api/api/views/media_views.py]: I meant mock a function so as to return the same argument back, something like

```python
mock_func = Mock(side_effect=lambda x: x)
```

This mock function, if called, will return its argument back.

File: api/test/test_auth.py [hunk 2]
```diff
@@ -206,21 +206,25 @@ def test_unauthed_response_headers(client):
...
+        mock_get_db_result.side_effect = lambda value: value
+        time.sleep(1)
+        token = test_auth_token_exchange["access_token"]
+        query_params = {
+            "unstable__sort_by": "indexed_on",
+            "unstable__sort_dir": sort_dir,
+        }
+        res = client.get(
+            "/v1/images/", query_params, HTTP_AUTHORIZATION=f"Bearer {token}"
+        )
```
Comment (dhruvkb) [inline api/test/test_auth.py:222]: Only these lines, where the view code is being executed, need to be inside the `patch` context, everything else can be at at the outer level as it was before.

**Review by dhruvkb — CHANGES_REQUESTED** [review]
I have small change, but otherwise it looks good to me. Thanks for  the contribution and for redoing the work based on reviews @ngken0995. The PR is almost ready to go 🚀!

File: api/test/test_auth.py (hunk 2 above)
Comment (ngken0995) [reply api/test/test_auth.py:223]: Thank you for trying to push me to understand testing.

**Review by AetherUnbound — APPROVED** [review]
These changes look great, thanks for making them @ngken0995!

**Review by dhruvkb — APPROVED** [review]
LGTM! This is great.

**Pull request merged**